test/e2e/cluster_samples_operator_test.go

```diff
@@ -33,24 +46,296 @@ func TestClusterSamplesOperator(t *testing.T) {
+			if err != nil {
+				t.Fatalf("%v", err)
+			}
+
+			streams, _ := names[imagestreamsKey]
+			streams[imagestream.Name] = true
+		}
+
+		if strings.HasSuffix(dir, templatesKey) {
+			template, err := h.Filetemplategetter.Get(dir + "/" + file.Name())
+			if err != nil {
+				t.Fatalf("%v", err)
+			}
+
+			templates, _ := names[templatesKey]
+			templates[template.Name] = true
+		}
+	}
+	return names
+}
+
+func verifyImagesPresent(t *testing.T, content map[string]bool) {
+	for key, _ := range content {
+		is := &imageapiv1.ImageStream{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ImageStream",
+				APIVersion: imageapiv1.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key,
+				Namespace: "openshift",
+			},
+		}
+
+		err := wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+			if err := sdk.Get(is); err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Fatalf("error waiting for example imagestream %s to appear: %v", key, err)
+		}
 	}
+}
+
+func verifyTemplatesPresent(t *testing.T, content map[string]bool) {
+	for key, _ := range content {
+		template := &templatev1.Template{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Template",
+				APIVersion: templatev1.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key,
+				Namespace: "openshift",
+			},
+		}
+
+		err := wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+			if err := sdk.Get(template); err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Errorf("error waiting for example template %s to appear: %v", key, err)
+		}
+	}
+}
+
+func validateContent(t *testing.T) {
+	contentDir := getContentDir(t)
+	content := getSamplesNames(contentDir, nil, t)
+	streams, _ := content[imagestreamsKey]
+	verifyImagesPresent(t, streams)
+	templates, _ := content[templatesKey]
+	verifyTemplatesPresent(t, templates)
+}
+
+func TestImageStreamInOpenshiftNamespace(t *testing.T) {
+	verifyOperatorUp(t)
+	validateContent(t)
+}
+
+func TestRecreateSamplesResourceAfterDelete(t *testing.T) {
+	sr := verifyOperatorUp(t)
+
+	oldTime := sr.CreationTimestamp
+
+	err := sdk.Delete(sr, sdk.WithDeleteOptions(&metav1.DeleteOptions{}))
+	if err != nil {
+		t.Errorf("error deleting samplesresource %v", err)
+	}
+
 	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
-		if err := sdk.Get(istag); err != nil {
+		if err := sdk.Get(sr); err != nil {
+			return false, nil
+		}
+		if sr.CreationTimestamp == oldTime {
 			return false, nil
 		}
 		return true, nil
 	})
 	if err != nil {
-		t.Errorf("error waiting for example imagestreamtag to appear: %v", err)
+		t.Errorf("creation times the same after delete: %v, %v, %#v", oldTime, sr.CreationTimestamp, sr)
+	}
+
+	err = verifyConditionsCompleteSamplesAdded(sr)
+	if err != nil {
+		t.Errorf("samples not re-established after delete %#v", sr)
 	}
```

Comment:
how are you confirming the samples ever got removed?

Reply:
The call the `validateContent` with the `&now` param below calls `verifyImagesPresent` and `verifyTemplatesPresent` with that same timestamp ... those method test the CreationTime of the samples they find to confirm they were recreated